src/components/Table/Table.Body.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react'
+
+import * as equal from 'fast-deep-equal'
+
+import Row from './Table.Row'
+
+import { BodyProps, BodyState } from './Table.types'
+
+export default class Body extends React.Component<BodyProps, BodyState> {
+  columnsCache = []
+  state: BodyState = { rows: [] }
+
+  constructor(props) {
+    super(props)
+    this.columnsCache = this.getColumnsCache(this.props.columns)
+
+    this.state.rows = this.getRows(props)
+  }
```

Comment:
@plbabin This can be written without `constructor`:

```jsx
export default class Body extends React.Component {
  state {
    rows: this.getRows(this.props)
  }
  columnsCache = this.getColumnsCache(this.props)

  ...
}
```

Reply:
nice I didn't know about that pattern!

Reply:
fixed in cfa50283